src/aiovodafone/api.py

```diff
@@ -255,10 +253,19 @@ async def login(self) -> bool:
             verify_ssl=False,
             allow_redirects=True,
         )
-        if reply.status != 200:
+        reply_json = await reply.json(content_type="text/html")
+        _LOGGER.debug("Login result: %s[%s]", LOGIN[int(reply_json)], reply_json)
+
+        if reply_json == "1":
+            return True
+
+        if reply_json == "2":
+            raise AlreadyLogged
+
+        if reply_json in ["3", "4"]:
```

Comment:
"3" and "4" mean "cannot authenticate: wrong credentials"
"5" means "cannot authenticate: retry later"

Reply:
"3" and "4" are correctly listed as "credential error"
"5" seems controversial based on the router type.